### ISSUE TITLE
Web component:  Leaving IBM - No label for the dialog ( both jaws and iPhone)

### ISSUE THREAD

**Issue by Praveen-sr** [carbon-design-system/carbon-for-ibm-dotcom#4596]
## Detailed description

Issue: No label for the dialog ( both jaws and iPhone)

Expected:
Provide label for the dialog by associating the Heading using aria-labelledby
4.1.2 Name, Role, Value

## Additional information

![image.png](https://images.zenhubusercontent.com/5e21a7b470118a498429a20e/5b24dca3-10a3-40c8-a80d-0864fed0b60c)
